source/Octopus.Tentacle.Kubernetes.Tests.Integration/KubernetesScriptServiceV1IntegrationTest.cs

```diff
@@ -46,6 +53,52 @@ sleep 0.1
         result.ExitCode.Should().Be(0);
         result.State.Should().Be(ProcessState.Complete);
 
+        recordedMethodUsages.For(nameof(IAsyncClientKubernetesScriptServiceV1.StartScriptAsync)).Started.Should().Be(1);
+        recordedMethodUsages.For(nameof(IAsyncClientKubernetesScriptServiceV1.GetStatusAsync)).Started.Should().BeGreaterThan(2).And.BeLessThan(30);
+        recordedMethodUsages.For(nameof(IAsyncClientKubernetesScriptServiceV1.CompleteScriptAsync)).Started.Should().Be(1);
+        recordedMethodUsages.For(nameof(IAsyncClientKubernetesScriptServiceV1.CancelScriptAsync)).Started.Should().Be(0);
+
+        return;
+
+        void StatusReceived(ScriptExecutionStatus status)
+        {
+            logs.AddRange(status.Logs);
+        }
+
+        Task ScriptCompleted(CancellationToken ct)
+        {
+            scriptCompleted = true;
+            return Task.CompletedTask;
+        }
+    }
+    
+    [Test]
+    public async Task SimpleScriptExitsWithErrorCode_ScriptFails()
+    {
+        // Arrange
+        var logs = new List<ProcessOutput>();
+        var scriptCompleted = false;
+
+        var command = new ExecuteKubernetesScriptCommandBuilder(LoggingUtils.CurrentTestHash())
+            .WithScriptBody(script => script
+                .Print("Hello World")
+                .ExitsWith(1))
+            .Build();
+
+        //act
+        var result = await TentacleClient.ExecuteScript(command, StatusReceived, ScriptCompleted, new InMemoryLog(), CancellationToken.None);
+
+        //Assert
+        logs.Should().Contain(po => po.Source == ProcessOutputSource.StdOut && po.Text == "Hello World");
+        scriptCompleted.Should().BeTrue();
+        result.ExitCode.Should().Be(1);
+        result.State.Should().Be(ProcessState.Complete);
+
+        recordedMethodUsages.For(nameof(IAsyncClientKubernetesScriptServiceV1.StartScriptAsync)).Started.Should().Be(1);
+        recordedMethodUsages.For(nameof(IAsyncClientKubernetesScriptServiceV1.GetStatusAsync)).Started.Should().BeGreaterThan(2).And.BeLessThan(30);
```

Comment:
Where do the numbers 2 and 30 come from? Is it based on an estimate given the polling nature of get status?
Perhaps it's sufficient to just assert that it's called rather than the number of times, in case of brittleness if the polling interval changes.

Reply:
I did. It wasn't serving much of a purpose